src/core/backends/etcd_metadata_backend.cpp

```
@@ -0,0 +1,285 @@
+        NIXL_ERROR << "Warning: Failed to remove etcd keys at " << key << " : "
+                   << response.error_message();
+        return NIXL_ERR_BACKEND;
+    }
+    catch (const std::exception &e) {
+        NIXL_ERROR << "Exception removing etcd keys at " << key << " : " << e.what();
+        return NIXL_ERR_BACKEND;
+    }
+}
+
+nixl_status_t
+nixlEtcdMetadataBackend::fetchOrWait(const std::string &agent_name,
```

Cppcheck (2.20.0)

[style] The function 'fetchOrWait' is never used.

(unusedFunction)